wrangler-core/src/test/java/io/cdap/directives/parser/CsvParserTest.java

```diff
@@ -67,4 +67,61 @@ public void testHeaders() throws Exception {
     Assert.assertEquals("alice", rows.get(0).getValue("first_name"));
     Assert.assertEquals("zed", rows.get(0).getValue("last_name"));
   }
+
+  @Test
+  public void testTrailingCommas() throws Exception {
+    String[] directives = new String[] {
+      "parse-as-csv body , false",
+      "filter-rows-on regex-match body_1 ^school_id$",
+      "drop body",
+      "set columns school_id, student_id, last_name, first_name",
+      "keep school_id,student_id,last_name,first_name"
+    };
+
+    List<Row> rows = Arrays.asList(
+      new Row("body", "school_id, student_id, last_name, first_name,,,"),
+      new Row("body", "14J456,33445566,Potter,Harry,,,"),
+      new Row("body", "14J456,44333433,Weasley,Ron,,,"),
+      new Row("body", "14J456,65765566,Granger,Hermione,,,"),
+      new Row("body", "14J456,13233121,Diggory,Cedric,,,"),
+      new Row("body", "14J456,98786868,Weasley,George,,,"),
+      new Row("body", "14J456,78977876,Weasley,Fred,,,")
+    );
+
+    rows = TestingRig.execute(directives, rows);
+    Assert.assertEquals(6, rows.size());
+    for (int i = 0; i < rows.size(); ++i) {
+      Assert.assertEquals(4, rows.get(i).width());
```

Comment:
tests should basically never just assert on a size. They should always verify the correct values are being generated.

Reply:
This is testing if all columns are correct or not. The intent of test was to check for right columns. anyway added each column test.